Martium.TravelInfo/Forms/TravelInfoForm.cs

```diff
@@ -57,17 +61,186 @@ private void InitializeMap()
+            DepartureCountryTextBox.Text = _travelInfoSettingsModel.DepartureCountry;
+            DepartureAddressTextBox.Text = _travelInfoSettingsModel.DepartureAddress;
+            PricePerKm.Text = _travelInfoSettingsModel.PricePerKm.ToString(CultureInfo.InvariantCulture);
+            AdditionalDistanceInKmTextBox.Text =
+                _travelInfoSettingsModel.AdditionalDistanceInKm.ToString(CultureInfo.InvariantCulture);
         }
 
+        private void DepartureCountryTextBox_TextChanged(object sender, System.EventArgs e)
+        {
+            if (DepartureCountryTextBox.Text == "LTU")
+            {
+                DepartureCountryTextBox.Text = "Lietuva"; // if you save data are this will not make error in database?
+            }
+        }
+
+        private void DepartureAddressTextBox_TextChanged(object sender, System.EventArgs e)
+        {
+            EnableSaveButton(DepartureAddressTextBox);
+            EnableSearchRouteButton();
+        }
+
+        private void ArrivalAddressTextBox_TextChanged(object sender, System.EventArgs e)
+        {
+            EnableSearchRouteButton();
+        }
+
+        private void PricePerKm_TextChanged(object sender, System.EventArgs e)
+        {
+            EnableSaveButton(PricePerKm);
+        }
+
+        private void AdditionalDistanceInKm_TextChanged(object sender, EventArgs e)
+        {
+            EnableSaveButton(AdditionalDistanceInKmTextBox);
+        }
+
+
+
         private void SetMapPositionByAddress(string address)
         {
             Map.SetPositionByKeywords(address);
         }
+
+        private void EnableSearchRouteButton()
+        {
+            SearchRouteButton.Enabled = (!string.IsNullOrWhiteSpace(DepartureAddressTextBox.Text) &&
+                                         !string.IsNullOrWhiteSpace(ArrivalAddressTextBox.Text));
+        }
+
+        private void EnableSaveButton(TextBox textBox) 
+        {
+            if (string.IsNullOrWhiteSpace(textBox.Text) || !CheckIsDouble(AdditionalDistanceInKmTextBox.Text) ||
+                !CheckIsDouble(PricePerKm.Text))
+            {
+              ControlAllSaveButtons(false);
+            }
+            else
+            {
+                ControlAllSaveButtons(true);
+            }
+        }
+
+        private bool CheckIsDouble(string text)
+        {
+            double success;
+            bool IsParsable = double.TryParse(text, out success);
+
+            return IsParsable;
+        }
+
+        private static void ShowInformationDialog(string message)
+        {
+            MessageBox.Show(message, "Info pranešimas", MessageBoxButtons.OK, MessageBoxIcon.Information);
+        }
+
+        private static void ShowErrorDialog(string message)
+        {
+            MessageBox.Show(message, "Klaidos pranešimas", MessageBoxButtons.OK, MessageBoxIcon.Error);
+        }
+
+
+
+        private void SaveDepartureAddressButton_Click(object sender, EventArgs e)
```

Comment:
event methods should be above private non event methods (move up to other event methods)

Reply:
same applies for methods below:
AdditionalDistanceInKmTextBox_Validating
PricePerKm_Validating
SavePricePerKmButton_Click
SaveAdditionalDistanceInKmButton_Click

Reply:
yeah i know this and was thinking to as you how make helpers methods? as was in funeral service

Reply:
google c# region 

but you don't need to create region just to change methods placements, region is optional thing 
my request was reorder methods so event methods would be first below constructor and only after them other private helper methods (if you want you can move helpers to region)